event-hubs.api.md

```diff
@@ -0,0 +1,303 @@
+## API Report File for "@azure/event-hubs"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AbortSignalLike } from '@azure/abort-controller';
+import { AmqpError } from 'rhea-promise';
+import { AwaitableSender } from 'rhea-promise';
+import { ConnectionContextBase } from '@azure/core-amqp';
+import { DataTransformer } from '@azure/core-amqp';
+import { DefaultDataTransformer } from '@azure/core-amqp';
+import { Dictionary } from 'rhea-promise';
+import { EventHubConnectionConfig } from '@azure/core-amqp';
+import { MessagingError } from '@azure/core-amqp';
+import { Receiver } from 'rhea-promise';
+import { ReceiverOptions } from 'rhea-promise';
+import { RetryOptions } from '@azure/core-amqp';
+import { SharedKeyCredential } from '@azure/core-amqp';
+import { Span } from '@azure/core-tracing';
+import { SpanContext } from '@azure/core-tracing';
+import { SpanOptions } from '@azure/core-tracing';
+import { TokenCredential } from '@azure/core-amqp';
+import { TokenType } from '@azure/core-amqp';
+import { WebSocketImpl } from 'rhea-promise';
+
+// @public
+export interface AbortSignalOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface Checkpoint {
+    consumerGroupName: string;
+    eTag: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    offset: number;
+    ownerId: string;
+    partitionId: string;
+    sequenceNumber: number;
+}
+
+// @public
+export interface CheckpointManager {
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export enum CloseReason {
+    OwnershipLost = "OwnershipLost",
+    Shutdown = "Shutdown"
+}
+
+// @public
+export interface CreateBatchOptions {
+    abortSignal?: AbortSignalLike;
+    maxSizeInBytes?: number;
+    partitionKey?: string;
+}
+
+export { DataTransformer }
+
+export { DefaultDataTransformer }
+
+// @public
+export interface EventData {
+    body: any;
+    properties?: {
+        [key: string]: any;
+    };
+}
+
+// @public
+export class EventDataBatch {
+    // Warning: (ae-forgotten-export) The symbol "ConnectionContext" needs to be exported by the entry point index.d.ts
+    // 
+    // @internal
+    constructor(context: ConnectionContext, maxSizeInBytes: number, partitionKey?: string);
+    readonly batchMessage: Buffer | undefined;
+    readonly count: number;
+    // @internal
+    readonly _messageSpanContexts: SpanContext[];
+    readonly partitionKey: string | undefined;
+    readonly sizeInBytes: number;
+    tryAdd(eventData: EventData, options?: TryAddOptions): boolean;
+}
+
+// @public
+export interface EventHubClientOptions {
+    dataTransformer?: DataTransformer;
```

Comment:
`DataTransformer` is an object with 2 properties `encode` and `decode` which are functions that are meant to be used on a message before sending and on a received message before passing it to the user.

Given that we have now split the clients for sending and receiving, does it make sense to have a "data transformer" anymore? Why not have `encode()` and `decode()` get passed to `EventHubProducerClientOptions` and `EventHubConsumerClientOptions` respectively?

Reply:
We're going to hide this interface for now - it's not common to all languages and it doesn't need to be an exposed feature until we're ready.